sdk/Sdk.Generators/FunctionMetadataProviderGenerator.Parser.cs

```diff
@@ -517,6 +517,29 @@ private bool TryGetAttributeProperties(AttributeData attributeData, Location att
                     }
                 }
 
+                // some properties have default values, so if these properties were not already defined in constructor or named arguments, we will auto-add them here
+                foreach (var member in attributeData.AttributeClass!.GetMembers().Where(a => a is IPropertySymbol))
+                {
+                    var defaultValAttrList = member.GetAttributes().Where(attr => SymbolEqualityComparer.Default.Equals(attr.AttributeClass, Compilation.GetTypeByMetadataName(Constants.DefaultValueType)));
+
+                    if (defaultValAttrList.SingleOrDefault() is { } defaultValAttr) // list will only be of size one b/c there cannot be duplicates of an attribute on one piece of syntax
+                    {
+                        var argName = member.Name;
+                        object arg = defaultValAttr.ConstructorArguments.SingleOrDefault().Value!; // only one constructor arg in DefaultValue attribute (the default value)
+                        if ( arg is bool b && string.Equals(argName, Constants.IsBatchedKey))
+                        {
+                            if (!attrProperties.Keys.Contains("Cardinality"))
+                            {
+                                attrProperties["Cardinality"] = b ? "Many" : "One";
+                            }
+                        }
+                        else if (!attrProperties.Keys.Any(a => string.Equals(a, argName, StringComparison.OrdinalIgnoreCase))) // check if this property has been assigned a value already in constructor or named args
```

Comment:
I noticed we actually own this dictionary. I am curious about why we sometimes want case sensitive and other times not case sensitive? I know I commented before about being case sensitive for a property check. Either way, if we can settle on a single correct way for key comparison, you can provide that comparer to the dictionary ctor:

``` CSharp
 Dictionary<string, object> attrProperties = new(StringComparer.OrdinalIgnoreCase);
```

Then you are free to use `attrProperties.ContainsKey({key})` and it will use the comparer from the ctor. Maybe ordinal ignore case is correct? But it might be more complex than that: when comparing a _property_ here, you want `Ordinal`, when comparing a _constructor parameter_, you want a comparator that will upper-case the first character only, then do an Ordinal comparison.

This is all getting into the weeds and may not be that important - so I'll leave it up to you to decide how accurate you want this to be. But give some thought to the scenario where a type here has both properties `MyProperty` and `Myproperty` (for reasons unknown to us) - will we consider them separately? How will we match a constructor parameter of `myProperty` vs `myproperty`?

Reply:
Yeah the non-case-sensitive check here is because we may have added the constructor parameter (which would be _isBatched_ while the property is _IsBatched_). It would be a stricter comparison to upper-case the first character of the constructor parameter and based on the convention we're following the end result should be the same. I guess it would also give attribute authors the flexibility to have `MyProperty` and `Myproperty` though I can't imagine why yet 😆. Leaning towards this change - will walk through it and test it out a bit first but if I do implement it I'll try to summarize the context we discussed here in the comments for future reference.